torch/distributed/algorithms/model_averaging/utils.py

```diff
@@ -0,0 +1,21 @@
+# flake8: noqa C101
+import torch
+import torch.distributed as dist
+
+
+def average_parameters(module: torch.nn.Module, process_group: dist.ProcessGroup):
+    """
+    Averages all the parameters of a given module.
+    For allreduce efficiency, all the parameters are flattened into a contiguous buffer.
+    Thus, it requires extra memory of the same size as the module's parameters.
+    """
+    group_to_use = process_group if process_group is not None else dist.group.WORLD
+
+    flat_params = torch.cat([p.data.view(-1) for p in module.parameters()])
+    flat_params /= dist.get_world_size(group_to_use)
```

Comment:
If rank is not in group what happens here? For example, if we have a subgroup of [0, 1] would rank 2 divide their params by 2 and then not average them?

Reply:
Actually I'm not sure if this will work properly if rank is not in group, as if rank is not in group and we call `get_world_size` we return `-1`: https://github.com/pytorch/pytorch/blob/master/torch/distributed/distributed_c10d.py#L824

So if the rank is not in group this might divide its parameters by -1? Can we test this?

Reply:
Good question. The input of any rank out of the subgroup will be divided by `-1`, so the result of the input `rank` will be `-rank`.